Integrate join event and event page

ssheloors opened the pull request:
Before copilot's incredible summary, a few things I noted down while doing this maybe we should discuss them: 
- The list of attendees does not refresh fast enough to show you after you register. I tried using ```useFocusEffect``` to refresh the data but it will only do it if you manually refocus the view or refresh this can be fixed later or we can change the approach of the list of attendees
- I added a button to cancel attendance. it doesn't do anything ofc since we haven't implemented that and we have it as a separate user story
- There should be a toast when successfully registering for the event. I think implementing toasts for messages should be a separate issue since we are also gonna use them for error handling

---
now copilot:

This pull request includes several changes to improve event handling and user interaction in the event pages. The most important changes include adding a router for navigation, updating the event join status, and redirecting users after event actions.

Enhancements to event handling and user interaction:

* `app/eventPage/[id].tsx`: Added `useRouter` for navigation and updated the event page to display join status and provide buttons for joining or canceling attendance. ([app/eventPage/[id].tsxL1-R1](diffhunk://#diff-81a951083e257f138cb65acfee6e678532015fa5f1537a80a8987626ec013a34L1-R1), [app/eventPage/[id].tsxL14-R22](diffhunk://#diff-81a951083e257f138cb65acfee6e678532015fa5f1537a80a8987626ec013a34L14-R22), [app/eventPage/[id].tsxR36-R48](diffhunk://#diff-81a951083e257f138cb65acfee6e678532015fa5f1537a80a8987626ec013a34R36-R48), [app/eventPage/[id].tsxL67-R82](diffhunk://#diff-81a951083e257f138cb65acfee6e678532015fa5f1537a80a8987626ec013a34L67-R82), [app/eventPage/[id].tsxR99-R132](diffhunk://#diff-81a951083e257f138cb65acfee6e678532015fa5f1537a80a8987626ec013a34R99-R132))
* [`app/joinEvent/attendeeDetails.tsx`](diffhunk://#diff-3dfe4ead7642e50da4fccba1ae8712f44cf3cc876383a6cc494cc60ad92c070bR13): Added `id` to local search params and updated the form submission to store join status in local storage and redirect to the event page. [[1]](diffhunk://#diff-3dfe4ead7642e50da4fccba1ae8712f44cf3cc876383a6cc494cc60ad92c070bR13) [[2]](diffhunk://#diff-3dfe4ead7642e50da4fccba1ae8712f44cf3cc876383a6cc494cc60ad92c070bL24-R36)
* [`app/joinEvent/index.tsx`](diffhunk://#diff-23beabf5b26df511a5ad836d9debd3fac09b2700e40f335b21748f8de8c2c584R39-R46): Modified the event creation process to redirect to the event page if the event is found.
* [`components/ui/IconSymbol.tsx`](diffhunk://#diff-c67a9a37839a6b851ff41aebac37315f85e03f67e610dee6084a1d2cea49d457R23): Added a new icon mapping for the "cross" symbol.

Minor changes:

* [`festivo-backend/src/app/(payload)/admin/importMap.js`](diffhunk://#diff-f848e3e38191f864e3aa0faf212f2b639fdd1860273f556d0129941ae054619dL1-R5): Reformatted the `importMap` export.

Closes #22

ssheloors committed:
Correct flow of pages and save attendance locally

ssheloors committed:
Remove unused imports

dotellie commented on app/eventPage/[id].tsx:
```diff
@@ -11,12 +11,15 @@ export default function EventPage() {
     return <Text>Loading...</Text>;
   }
 
+  const joinStatus = localStorage.getItem(event.eventCode);
```
This can't be local storage! Use the `useStorage` hook instead

ssheloors committed:
Merge branch 'main' into 22-integrate-join-event-and-event-page

ssheloors committed:
Adjust merge conflict

dotellie commented on app/eventPage/[id].tsx:
```diff
@@ -66,23 +81,56 @@ export default function EventPage() {
+        typeof event.organizer !== "number" &&
+        user?.id !== event.organizer.id && (
+          <FloatingActionButton
+            iconAfter={
```
Instead of attaching a listener here, use the `Link` component together with the button as we have done in other places

ssheloors replied:
Is there a specific reason for this? or just preference because in my opinion, it looks way more cluttered :/ or am I missing another way of passing the params?

```js    
<Link
push
href={`/joinEvent/attendeeDetails?code=${encodeURIComponent(event.eventCode)}&id=${encodeURIComponent(event.id)}`}
> 
````

dotellie replied:
The main reason is accessibility and semantics. I don't know if the snippet you showed is the only way to go about it, but yeah the Link component should be used whenever possible!

ssheloors committed:
run prettier

dotellie reviewed (changes requested):
Quick review on my phone, will take a deeper look tomorrow!

ssheloors committed:
Requested changes

ssheloors committed:
prettier

dotellie commented on app/eventPage/[id].tsx:
```diff
@@ -72,17 +93,48 @@ export default function EventPage() {
+              iconAfter={
+                <IconSymbol name="arrow.right" size={24} color="white" />
+              }
+              color="white"
```
Haven't looked at how this turns out when running the app, but my hunch says that this should probably be `theme="accent"` instead 😅

dotellie commented on app/joinEvent/attendeeDetails.tsx:
```diff
@@ -33,13 +34,14 @@ export default function AttendeeDetails() {
-      router.push({
-        pathname: "/joinEvent/success",
-        params: { code: code },
+      localStorage.setItem(code, "joined");
```
This also needs to be replaced with the storage thing!

dotellie commented on app/eventPage/[id].tsx:
```diff
@@ -1,18 +1,27 @@
-  // const navigation = useNavigation();
+
+  const storage = useStorage();
+  const [joinStatus, setJoinStatus] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (event) {
+      storage.getString(event.eventCode).then(setJoinStatus);
+    }
+  }, [event, storage]);
```
This works, but ideally I would see react query be used here so that we have all the different statuses and cache in place

dotellie commented on app/joinEvent/attendeeDetails.tsx:
```diff
@@ -20,6 +20,7 @@ const formSchema = z.object({
 
 export default function AttendeeDetails() {
   const { code } = useLocalSearchParams() as { code: string };
+  const { id } = useLocalSearchParams() as { id: string };
```
This feels kind of strange, no? Both identifies the same event so I feel like we can get rid of one and later fetch the other.

dotellie commented on app/eventPage/[id].tsx:
```diff
@@ -72,17 +93,48 @@ export default function EventPage() {
+      {joinStatus !== "joined" &&
+        typeof event.organizer !== "number" &&
+        user?.id !== event.organizer.id && (
+          <Link
+            push
+            href={`/joinEvent/attendeeDetails?code=${encodeURIComponent(event.eventCode)}&id=${encodeURIComponent(event.id)}`}
+          >
```
I found a nicer way to do this! You can [use the `href` as an object and then add params](https://docs.expo.dev/router/navigating-pages/#linking-to-dynamic-routes) like this:

```tsx
      <Link
        href={{
          pathname: '/user/[id]',
          params: { id: 'bacon' }
        }}>
          View user
        </Link>
```

dotellie commented on app/eventPage/[id].tsx:
```diff
@@ -72,17 +93,48 @@ export default function EventPage() {
+      {joinStatus !== "joined" &&
+        typeof event.organizer !== "number" &&
+        user?.id !== event.organizer.id && (
+          <Link
```
This button renders weirdly

<img width="555" alt="image" src="https://github.com/user-attachments/assets/77ed2b6c-4298-46ac-8d36-9b26c5159f7e">

I think I know why though - this link is missing an `asChild` prop which would make it so that the button inside is the link instead of the link containing the button (and yeah, I know, that's how it is supposed to work on the web but RN is somewhat different in that aspect I suppose)

dotellie reviewed (changes requested):
Looking good so far! 🌟  The only thing that absolutely needs to be fixed before merging is the usage of localStorage, but otherwise my comments are just suggestions for improvements 😊

ssheloors committed:
Change usage of storage and Link

ssheloors committed:
prettier

ssheloors commented:
I will just summarize here instead of replying 1 by 1,
I changed the storage usage and the Link path and Button since they are the most critical things, in terms of the hook, I would rather not touch it because I don't know exactly how to do it so I would rather not spend more time on this. 

I added the prop to the button but I can't test if it looks good now because I can't get to that page in the emulator (same reason why I missed it xd) could you try it when you have time?

In terms of the accent color, I was trying out white and I think it looks wayyy better but if we make that change it has to be in the global theme, I know. We can talk abt it once we are going through everything

You can merge if the button looks good so that we can move on, thanks fo the review 🔥

dotellie reviewed (approved):
Looks somewhat wonky when you're registered, but you're right, nothing to block over!

<img width="555" alt="image" src="https://github.com/user-attachments/assets/fe91e58f-759a-4550-a026-dd85dfe4d8ac">

LGTM!

dotellie commented:
I guess you did say to merge if I wanted, so I'll go ahead and do that so I can add some tests here 😄
